lensless/benchmark.py

```diff
@@ -0,0 +1,192 @@
+        Reconstruction algorithm to benchmark
+    data : str
+        Path to the test dataset.
+        It is expected to contained a file psf.tiff and two folder :
+          diffuser and lensed containing each pair of test image (lensless and lensed) with the same name as a .npy file.
+    downsample : int, optional
+        By how mush the psf and image should be downsample, by default 4
+    n_files : int, optional
+        Metrics will be computed only on the first n_files images, by default 100
+    batchsize : int, optional
+        Batch size for processing. For maximum compatibility use 1 (batchsize above 1 are not supported on all algorithm), by default 1
+
+    Returns
+    -------
+    Dict[str, float]
+        A dictionnary containing the metrics name and average value
+    """
+    assert isinstance(model._psf, torch.Tensor), "model need to be constructed with torch support"
+    device = model._psf.device
+
+    psf_fp = os.path.join(data, "psf.tiff")
+    psf_float, background = load_psf(
+        psf_fp,
+        downsample=downsample,
+        return_float=True,
+        return_bg=True,
+        bg_pix=(0, 15),
+    )
+    background = torch.from_numpy(background)
+
+    dataset = RealDataset(data, n_files=n_files, background=background)
+    dataloader = DataLoader(dataset, batch_size=batchsize, pin_memory=(device != "cpu"))
+
+    metrics = {
+        "MSE": MSELoss().to(device),
+        "MAE": L1Loss().to(device),
+        "LPIPS": lpip.LearnedPerceptualImagePatchSimilarity(net_type="vgg").to(device),
+        "PSNR": psnr.PeakSignalNoiseRatio().to(device),
+        "SSIM": StructuralSimilarityIndexMeasure().to(device),
+    }
+    metrics_values = {key: 0.0 for key in metrics}
+
+    for diffuser, lensed in tqdm(dataloader):
+        diffuser = diffuser.to(device).squeeze()
+        lensed = lensed.to(device).permute(0, 3, 1, 2)
+
+        # compute predictions
+        with torch.no_grad():
+            if batchsize == 1:
+                model.set_data(diffuser)
+                prediction = model.apply(plot=False, save=False, **kwargs)[None, :, :, :].permute(
+                    0, 3, 1, 2
+                )
+            else:
+                prediction = model.batch_call(plot=False, save=False, **kwargs).permute(0, 3, 1, 2)
+
+        # normalization
+        prediction_max = torch.amax(prediction, dim=(1, 2, 3), keepdim=True)
+        prediction = prediction / prediction_max
+        lensed_max = torch.amax(lensed, dim=(1, 2, 3), keepdim=True)
+        lensed = lensed / lensed_max
+        # compute metrics
+        for metric in metrics:
+            metrics_values[metric] += metrics[metric](prediction, lensed).cpu().item()
+
+    # average metrics
+    for metric in metrics:
+        metrics_values[metric] /= len(dataloader)
+
+    return metrics_values
+
+
+if __name__ == "__main__":
+    from lensless import ADMM
+
+    downsample = 4
+    device = "cpu"
+
+    data = "data/DiffuserCam_Mirflickr"
```

Comment:
could you change it to `data/DiffuserCam_Mirflickr_200_3011302021_11h43_seed11` to correspond to the subset from [here](https://drive.switch.ch/index.php/s/vmAZzryGI8U8rcE)?
and perhaps add a check just after to download the dataset and put in `data` (asking user to confirm) if it's not already there?

Reply:
https://pythonguides.com/download-zip-file-from-url-using-python/